core/lib/snapshots_applier/src/lib.rs

```diff
@@ -588,18 +588,27 @@ impl<'a> SnapshotsApplier<'a> {
             factory_deps.factory_deps.len()
         );
 
-        let all_deps_hashmap: HashMap<H256, Vec<u8>> = factory_deps
-            .factory_deps
-            .into_iter()
-            .map(|dep| (hash_bytecode(&dep.bytecode.0), dep.bytecode.0))
-            .collect();
-        storage
-            .factory_deps_dal()
-            .insert_factory_deps(
-                self.applied_snapshot_status.l2_block_number,
-                &all_deps_hashmap,
-            )
-            .await?;
+        // we cannot insert all factory deps because of field size limit triggered by UNNEST
+        // in underlying query, see `https://www.postgresql.org/docs/current/limits.html`
+        // there were around 100 thousand contracts on mainnet, where this issue first manifested
+        let chunk_size = 1000;
+        let chunks_count = factory_deps.factory_deps.len().div_ceil(chunk_size);
+        for chunk_id in 0..chunks_count {
+            let chunk_start = chunk_id * chunk_size;
+            let chunk_end = min(factory_deps.factory_deps.len(), (chunk_id + 1) * chunk_size);
+            let chunk = &factory_deps.factory_deps[chunk_start..chunk_end];
```

Comment:
I think it's possible to use [standard slice chunking](https://doc.rust-lang.org/std/primitive.slice.html#method.chunks) here; would be shorter and more idiomatic.

Reply:
nice!